### PR TITLE
Pet charity be feature/animal schema

### PR DESCRIPTION
**Pull request by alexander-svirzhevsky**

### CODE REVIEW

File: db/Schema/Breed.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require("mongoose");
+
+const Breed = new mongoose.Schema({
+  breedName: {
+    type: String,
+    required: true,
+  },
+  type: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "typeOfAnimal",
```
Comment (Umalishenni) [inline db/Schema/Breed.js:10]: To be honest, AnimalType sounds much better.

File: routes/api/animal.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+const Animal = require("../../db/Schema/Animal");
+const Breed = require("../../db/Schema/Breed");
+
+router.post(
+  "/",
+  [
+    check("name", "Name is required").not().isEmpty(),
+    check("sex", "Sex is required").not().isEmpty(),
+    check("type", "Type is required").not().isEmpty(),
+    check("breedName", "breedName is required").not().isEmpty(),
```
Comment (Umalishenni) [inline routes/api/animal.js:15]: where is validation for "age"?

File: routes/api/animal.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+      const { name, age, sex, type, breedName } = req.body;
+
+      const typeId = await TypeOfAnimal.findOne({ type: type });
+      const breedId = await Breed.findOne({ breedName: breedName });
```
Comment (Umalishenni) [inline routes/api/animal.js:27]: what will happen if you don't find type or breed? You can't save a document without those properties, write cases for it

File: server.js [hunk 4]
```diff
@@ -5,12 +5,17 @@ const connectDB = require("./db/db");
 
 connectDB();
 
+const Animals = require("./db/Schema/Animal");
```
Comment (Umalishenni) [inline server.js:8]: why do you need this here?

File: routes/api/animal.js (hunk 2 above)
Comment (alexander-svirzhevsky) [reply routes/api/animal.js]: Sometimes people in my organization won't know how old is some animal

**Pull request merged**